src/gravel/controllers/nodes/local.py

```diff
@@ -0,0 +1,185 @@
+        qualified = False
+        error = (
+            "The node does not have a sufficient number of CPU cores. "
+            "Required: %d, Actual: %d." % (min_cpu, actual_cpu)
+        )
+        status = HardwareQualifiedEnum.ERROR
+
+    return CPUQualifiedModel(
+        qualified=qualified,
+        min_cpu=min_cpu,
+        actual_cpu=actual_cpu,
+        error=error,
+        status=status,
+    )
+
+
+async def validate_memory() -> MemoryQualifiedModel:
+    """
+    Validates the localhost meets the minium memory requirements.
+    """
+    qualified: bool = True
+    min_mem: int = AQUARIUM_MIN_MEMORY_GB
+    # 1024 kb / 1024 mb / 1024 gb
+    # NOTE(jhesketh): We round down to the nearest GB
+    actual_mem: int = int(psutil.virtual_memory().total / 1024 / 1024 / 1024)
+    error: str = ""
+    status: HardwareQualifiedEnum = HardwareQualifiedEnum.QUALIFIED
+
+    if actual_mem < min_mem:
+        qualified = False
+        error = (
+            "The node does not have a sufficient memory. "
+            "Required: %d, Actual: %d." % (min_mem, actual_mem)
+        )
+        status = HardwareQualifiedEnum.ERROR
+    return MemoryQualifiedModel(
+        qualified=qualified,
+        min_mem=min_mem,
+        actual_mem=actual_mem,
+        error=error,
+        status=status,
+    )
+
+
+async def validate_root_disk() -> RootDiskQualifiedModel:
+    """
+    Validates the localhost meets the minium disk requirements.
+
+    NOTE: This only verifies the total size of the root partition. It does
+          not validate the amount of free space etc.
+    """
+    qualified: bool = True
+    min_disk: int = AQUARIUM_MIN_ROOT_DISK_GB
+    # 1024 kb / 1024 mb / 1024 gb
+    # NOTE(jhesketh): We round down to the nearest GB
+    actual_disk: int = int(psutil.disk_usage("/").total / 1024 / 1024 / 1024)
+    error: str = ""
+    status: HardwareQualifiedEnum = HardwareQualifiedEnum.QUALIFIED
+
+    if actual_disk < min_disk:
+        qualified = False
+        error = (
+            "The node does not have sufficient space on the root disk. "
+            "Required: %d, Actual: %d." % (min_disk, actual_disk)
+        )
+        status = HardwareQualifiedEnum.ERROR
+    return RootDiskQualifiedModel(
+        qualified=qualified,
+        min_disk=min_disk,
+        actual_disk=actual_disk,
+        error=error,
+        status=status,
+    )
+
+
+async def localhost_qualified() -> LocalhostQualifiedModel:
+    """
+    Validates whether the localhost is fully qualified (ie, meets all minium
+    requirements).
+    """
+    all_qualified: bool = True
+    errors: List = []
+
+    cputask = asyncio.create_task(validate_cpu())
+    memtask = asyncio.create_task(validate_memory())
+    disktask = asyncio.create_task(validate_root_disk())
+
+    cpu_qualified: CPUQualifiedModel = await cputask
+    mem_qualified: MemoryQualifiedModel = await memtask
+    root_disk_qualified: RootDiskQualifiedModel = await disktask
```

Comment:
What about using this?
```
cpu_qualified: CPUQualifiedModel, mem_qualified: MemoryQualifiedModel, root_disk_qualified: RootDiskQualifiedModel = await asyncio.gather(validate_cpu(), validate_memory(), validate_root_disk())
```
OK, it will not reduce code lines, but complexity.

Reply:
that's a really good idea

Reply:
Yep, agreed.